### PR TITLE
(HLT) fix service discovery in HLTriggerJSONMonitoring

### PR DESCRIPTION
**Pull request by smorovic**
#### PR description:

Service template type should not be a pointer. This caused FastMonitoringService service to not be detected and caused output not being suppressed when exception is thrown in HLT.

#### PR validation:

Modification was tested on replayed error stream data from 2024 HI run and fixes the problem.

### CODE REVIEW

**Comment by smorovic** [comment]
@cmsbuild please test

File: HLTrigger/JSONMonitoring/plugins/HLTriggerJSONMonitoring.cc [hunk 1]
```diff
@@ -430,9 +430,9 @@ void HLTriggerJSONMonitoring::globalEndLuminosityBlockSummary(edm::LuminosityBlo
   unsigned int run = lumi.run();
 
   bool writeFiles = true;
-  if (edm::Service<evf::FastMonitoringService*>().isAvailable()) {
+  if (edm::Service<evf::FastMonitoringService>().isAvailable()) {
     evf::FastMonitoringService* fms =
-        (evf::FastMonitoringService*)(edm::Service<evf::FastMonitoringService*>().operator->());
+        (evf::FastMonitoringService*)(edm::Service<evf::FastMonitoringService>().operator->());
```
Comment (fwyzard) [inline HLTrigger/JSONMonitoring/plugins/HLTriggerJSONMonitoring.cc:435]: can the cast be removed ?
```c++
    evf::FastMonitoringService* fms = edm::Service<evf::FastMonitoringService>().operator->();
```

Comment (fwyzard) [reply HLTrigger/JSONMonitoring/plugins/HLTriggerJSONMonitoring.cc:435]: the whole block could also be rewritten more simply as
```c++
  if (edm::Service<evf::FastMonitoringService>().isAvailable()) {
    writeFiles = edm::Service<evf::FastMonitoringService>()->shouldWriteFiles(ls);
  }
```

**Comment by mmusich** [comment]
+hlt

**Comment by mandrenguyen** [comment]
+1

**Pull request merged**